openshift-knative-operator/pkg/common/config.go

```diff
@@ -34,3 +34,22 @@ func ConfigureIfUnset(s *base.CommonSpec, cm, key, value string) {
 	}
 	s.Config[cm][key] = value
 }
+
+// ConfigureIfUnsetAny sets a value in the config-domain if any configuration is not already set.
+// For example, the config-domain can take an arbitrary domain as a key, so it should be used here.
+func ConfigureIfUnsetAny(s *base.CommonSpec, cm, key, value string) {
+	if s.Config == nil {
+		s.Config = make(map[string]map[string]string, 1)
+	}
+
+	if s.Config[cm] == nil {
+		s.Config[cm] = make(map[string]string, 1)
+	}
+
+	if len(s.Config[cm]) != 0 {
```

Comment:
Just to be clear in that domain cm there can be only one entry?

Reply:
I don't thinks so, you can have a list of domains. @nak3 can you maybe add a test case where one key has multiple values in its map? like

domain:
  example.com: xxx
  second.com: xxx